src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -22,6 +22,10 @@ public class Dictionary<TKey, TValue> : IDictionary<TKey, TValue>, IDictionary,
         private const string KeyValuePairsName = "KeyValuePairs"; // Do not rename (binary serialization)
         private const string ComparerName = "Comparer"; // Do not rename (binary serialization)
 
+#pragma warning disable CA1825 // avoid the extra generic instantiation for Array.Empty<Entry>()
+        private static readonly Entry[] s_emptyArray = new Entry[0];
```

Comment:
Isn't this still subject to multiple instantiation given its size varies with TKey and TValue? (@jkotas?)

Reply:
The concern is the extra instantiations of `Array.Empty`.

Reply:
Right, so that's avoided, but every instantiation of D<K,V> will have an extra static field here (and array). Is that less impactful?

Reply:
Right. It is still going to be created for each Dictionary, but with less overhead than `Array.Empty`.

Reply:
> Is that less impactful?

Yes, it is less impactful.

Reply:
Is similar to what `List<T>` does https://github.com/dotnet/runtime/blob/ec5d2eeade1ba4f203b2166c399056e124b4cfb8/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs#L29-L31